build images occur  error

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Dify version

1.14.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

I pull the source code, then build the image docker build -t dify_web:20260527 -f web/Dockerfile in the root directory.
Tip this
(base) root@dfwy-server:/opt/dify-1.14.2# docker build -t dfwy_design_web:20260527 -f web/Dockerfile .
[+] Building 4.1s (17/26)                                                                                                        docker:default
 => [internal] load build definition from Dockerfile                                                                                       0.0s
 => => transferring dockerfile: 3.01kB                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/node:22-alpine                                                                          0.0s
 => [base 1/3] FROM docker.io/library/node:22-alpine                                                                                       0.0s
 => [internal] load build context                                                                                                          0.7s
 => => transferring context: 842.37kB                                                                                                      0.7s
 => CACHED [base 2/3] RUN apk add --no-cache tzdata                                                                                        0.0s
 => CACHED [base 3/3] RUN corepack enable                                                                                                  0.0s
 => CACHED [packages 1/8] WORKDIR /app                                                                                                     0.0s
 => CACHED [production 1/8] RUN ln -s /usr/share/zoneinfo/UTC /etc/localtime     && echo UTC > /etc/timezone                               0.0s
 => CACHED [production 2/8] RUN addgroup -S -g 1001 dify &&     adduser -S -u 1001 -G dify -s /bin/ash -h /home/dify dify &&     mkdir /a  0.0s
 => CACHED [production 3/8] WORKDIR /app                                                                                                   0.0s
 => CACHED [packages 2/8] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml /app/                                                       0.0s
 => CACHED [packages 3/8] COPY web/package.json /app/web/                                                                                  0.0s
 => CACHED [packages 4/8] COPY e2e/package.json /app/e2e/                                                                                  0.0s
 => CACHED [packages 5/8] COPY sdks/nodejs-client/package.json /app/sdks/nodejs-client/                                                    0.0s
 => CACHED [packages 6/8] COPY packages /app/packages                                                                                      0.0s
 => CACHED [packages 7/8] RUN corepack install                                                                                             0.0s
 => ERROR [packages 8/8] RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile                                                               3.3s
------                                                                                                                                          
 > [packages 8/8] RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile:                                                                          
3.230 [ERR_PNPM_UNSUPPORTED_ENGINE] Unsupported environment (bad pnpm and/or Node.js version)                                                   
3.230                                                                                                                                           
3.230 Your Node version is incompatible with "/app".                                                                                            
3.230 
3.230 Expected version: ^22.22.1
3.230 Got: v22.14.0
3.230 
3.230 This is happening because the package's manifest has an engines.node field specified.
3.230 To fix this issue, install the required Node version.
------
Dockerfile:34
--------------------
  32 |     
  33 |     # Install workspace dependencies for the Docker build environment.
  34 | >>> RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile
  35 |     
  36 |     # build resources
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile" did not complete successfully: exit code: 1
(base) root@dfwy-server:/opt/dify-1.14.2# 


### ✔️ Expected Behavior

I pull the source code, then build the image docker build -t dify_web:20260527 -f web/Dockerfile in the root directory.
Tip this
(base) root@dfwy-server:/opt/dify-1.14.2# docker build -t dfwy_design_web:20260527 -f web/Dockerfile .
[+] Building 4.1s (17/26)                                                                                                        docker:default
 => [internal] load build definition from Dockerfile                                                                                       0.0s
 => => transferring dockerfile: 3.01kB                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/node:22-alpine                                                                          0.0s
 => [base 1/3] FROM docker.io/library/node:22-alpine                                                                                       0.0s
 => [internal] load build context                                                                                                          0.7s
 => => transferring context: 842.37kB                                                                                                      0.7s
 => CACHED [base 2/3] RUN apk add --no-cache tzdata                                                                                        0.0s
 => CACHED [base 3/3] RUN corepack enable                                                                                                  0.0s
 => CACHED [packages 1/8] WORKDIR /app                                                                                                     0.0s
 => CACHED [production 1/8] RUN ln -s /usr/share/zoneinfo/UTC /etc/localtime     && echo UTC > /etc/timezone                               0.0s
 => CACHED [production 2/8] RUN addgroup -S -g 1001 dify &&     adduser -S -u 1001 -G dify -s /bin/ash -h /home/dify dify &&     mkdir /a  0.0s
 => CACHED [production 3/8] WORKDIR /app                                                                                                   0.0s
 => CACHED [packages 2/8] COPY package.json pnpm-lock.yaml pnpm-workspace.yaml /app/                                                       0.0s
 => CACHED [packages 3/8] COPY web/package.json /app/web/                                                                                  0.0s
 => CACHED [packages 4/8] COPY e2e/package.json /app/e2e/                                                                                  0.0s
 => CACHED [packages 5/8] COPY sdks/nodejs-client/package.json /app/sdks/nodejs-client/                                                    0.0s
 => CACHED [packages 6/8] COPY packages /app/packages                                                                                      0.0s
 => CACHED [packages 7/8] RUN corepack install                                                                                             0.0s
 => ERROR [packages 8/8] RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile                                                               3.3s
------                                                                                                                                          
 > [packages 8/8] RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile:                                                                          
3.230 [ERR_PNPM_UNSUPPORTED_ENGINE] Unsupported environment (bad pnpm and/or Node.js version)                                                   
3.230                                                                                                                                           
3.230 Your Node version is incompatible with "/app".                                                                                            
3.230 
3.230 Expected version: ^22.22.1
3.230 Got: v22.14.0
3.230 
3.230 This is happening because the package's manifest has an engines.node field specified.
3.230 To fix this issue, install the required Node version.
------
Dockerfile:34
--------------------
  32 |     
  33 |     # Install workspace dependencies for the Docker build environment.
  34 | >>> RUN VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile
  35 |     
  36 |     # build resources
--------------------
ERROR: failed to build: failed to solve: process "/bin/sh -c VITE_GIT_HOOKS=0 pnpm install --frozen-lockfile" did not complete successfully: exit code: 1
(base) root@dfwy-server:/opt/dify-1.14.2# 


### ❌ Actual Behavior

1

--- a/web/Dockerfile
+++ b/web/Dockerfile
@@ -1,5 +1,5 @@
 # base image
-FROM node:22-alpine AS base
+FROM node:22.22.1-alpine AS base
 LABEL maintainer="takatost@gmail.com"
 
 # if you located in China, you can use aliyun mirror to speed up

